polytorch/data.py

```diff
@@ -1,19 +1,27 @@
 from torch import nn
+from typing import List
 from dataclasses import dataclass
 
 from .enums import ContinuousDataLossType
 
 class PolyData():
+    name:str = ""
+
     def create_module(self, embedding_size:int):
         raise NotImplementedError(f'Use either CategoricalData, OrdinalData or ContinuousData')
 
     def size(self) -> int:
         raise NotImplementedError(f'Use either CategoricalData, OrdinalData or ContinuousData')
-    
+
+    def get_name(self) -> str:
+        return self.name or self.__class__.__name__
+
 
 @dataclass
 class CategoricalData(PolyData):
     category_count:int
+    labels:List[str] = None
```

Comment:
These would seem to need a `| None` in the type definition?

Reply:
yes - but I didn't want to add future_typing as a dependency so i was lazy here. I think I'll add it.

Reply:
Or `Optional[List[str]]` in that case? With dataclasses being one of the only cases where type hints are used at runtime, seems worth defining (it seems happy enough though, which surprised me).

Reply:
yeah - much better